sdk/python/feast/infra/online_stores/contrib/hazelcast_online_store/hazelcast_online_store.py

```diff
@@ -0,0 +1,252 @@
+
+        client = self._get_client(online_store_config)
+        fv_map = client.get_map(_table_id(config.project, table))
+
+        for entity_key, values, event_ts, created_ts in data:
+            entity_key_bin = serialize_entity_key(
+                entity_key,
+                entity_key_serialization_version=config.entity_key_serialization_version,
+            ).hex()
+            event_ts_utc = _to_utc_timestamp(event_ts)
+            created_ts_utc = 0.0
+            if created_ts is not None:
+                created_ts_utc = _to_utc_timestamp(created_ts)
+            for feature_name, value in values.items():
+                feature_value = value.SerializeToString().hex()
+                __key = str(entity_key_bin) + feature_name
+                fv_map.put(
+                    __key,
+                    HazelcastJsonValue(
+                        {
+                            "entity_key": entity_key_bin,
+                            "feature_name": feature_name,
+                            "feature_value": feature_value,
+                            "event_ts": event_ts_utc,
+                            "created_ts": created_ts_utc,
+                        }
+                    ),
+                    config.online_store.key_ttl_seconds,
+                )
+                if progress:
+                    progress(1)
+
+    def online_read(
+        self,
+        config: RepoConfig,
+        table: FeatureView,
+        entity_keys: List[EntityKeyProto],
+        requested_features: Optional[List[str]] = None,
+    ) -> List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]]:
+        online_store_config = config.online_store
+        assert isinstance(online_store_config, HazelcastOnlineStoreConfig)
+
+        client = self._get_client(online_store_config)
+        entries: List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]] = []
+        fv_map = client.get_map(_table_id(config.project, table))
+
+        hz_keys = []
+        entity_keys_str = {}
+        for entity_key in entity_keys:
+            feature_keys = []
+            entity_key_str = str(
+                serialize_entity_key(
+                    entity_key,
+                    entity_key_serialization_version=config.entity_key_serialization_version,
+                ).hex()
+            )
+            if requested_features:
+                for feature in requested_features:
+                    feature_keys.append(entity_key_str + feature)
+            else:
+                for f in table.features:
+                    feature_keys.append(entity_key_str + f.name)
+            hz_keys.extend(feature_keys)
+            entity_keys_str[entity_key_str] = feature_keys
+
+        data = fv_map.get_all(hz_keys).result()
+        entities = []
+        for _key in hz_keys:
+            try:
+                data[_key] = data[_key].loads()
+                entities.append(data[_key]["entity_key"])
+            except KeyError:
```

Comment:
Is it possible to get `KeyError`?

Reply:
Actually one of the tests were failing because of this and I had to add this check.